server/opts.go

```diff
@@ -562,6 +562,14 @@ type MQTTOpts struct {
 
 	// Snapshot of configured TLS options.
 	tlsConfigOpts *TLSConfigOpts
+
+	// rejectQoS2Pub tells the MQTT client to not accept QoS2 PUBLISH, instead
+	// error and terminate the connection.
+	rejectQoS2Pub bool
```

Comment:
Unless this is temporary, the option names need to start with a upper-case: `RejectQoS2Pub` to be exported, so that users can set this option when importing NATS Server code (embedding it). It works for you because the test belongs to the same `server` package. It would not from the `test` package for instance because these fields would then not be visible.

Reply:
This was deliberate - I am considering this a temporary fix, we can always make it public and documented later if we decide it worthwhile.

Reply:
Ok, understood.